src/garage/tf/models/base.py

```diff
@@ -441,3 +445,216 @@ def __setstate__(self, state):
         """
         self.__dict__.update(state)
         self._networks = {}
+
+
+class Module(abc.ABC):
```

Comment:
thank you for doing this -- gets us one step closer to moving to `tf.Module`

Reply:
What's the most distinct difference between a model and a module?

Reply:
@ahtsan Module is built on top of Model. Policy, Regressor, Baseline are all Modules.

Reply:
As I state in my comment below, it's clear there will be significant refactoring for all of these classes in the near future, so I'd like to avoid a lengthy design discussion now.

@naeioi 's PR make a significant improvement (collecting a lot of repetitive code) towards unifying the TF primitive APIs, so I'm happy to merge it right now, even if it's not necessarily globally-optimal.

If you want a better and more careful design for the TF primitive tree, the best thing you can do is get TE merged, so we can cut the release, so we can start working on the API unification.

See https://github.com/rlworkgroup/garage/milestone/14 for what I plan on having us work on as soon as the release is cut. Notice any themes?

Reply:
Cool. Once I have the CI ready I will merge this.

Reply:
@ryanjulian Sounds good.
@naeioi Great work!